### PR TITLE
feat: convert RichTrack factory to JOmniFactory

### PR DESCRIPTION
**Pull request by wdconinc**
### Briefly, what does this PR introduce?
This PR converts the RichTrack_factory to JOmniFactory. This is one of the holdouts for this conversion since it does so much in terms of setting up planes, including parsing of collection names. Well, that's a thing of the past now.
- RichTrack_factory is actually a TrackPropagation_factory and could likely be the same factory (descoped).
- The configuration (tracking and filter surfaces) is now created in the plugin.
- TrackPropagation algorithm now understands about filter surfaces in its configuration.
- RichGeo_service's ActsGeo now returns tracking and filter surfaces instead of Acts surfaces.

What is lost compared to previously?
- `eicrecon -PDRICH:gas:numPlanes=10` won't work anymore and this needs to be provided through the plugin. If really required, this can be brought back since `DRICH::InitPlugin` has access to the `JApplication*` pointer, but I'd rather not set a precedent that it's ok to define command line arguments there...
- This now runs track propagation for DRICH tracks twice: once in the context of aerogel filtering and once for gas filtering. This all needs a more comprehensive re-evaluation that we are currently not ready for, though. Ultimately we need 1 track propagation to all surfaces, and then a distribution of those results to algorithms. Acts is not ready for that, nor are we.

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [x] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
Yes, if you were relying on the command line parameter `DRICH:gas:numPlanes=10`.

### Does this PR change default behavior?
Yes, if you were relying on the command line parameter `DRICH:gas:numPlanes=10`.

### CODE REVIEW

**Comment by wdconinc** [comment]
Asked for early reviews, even though this is currently killing the DRICH reconstruction, but that's just a temporary issue.

**Comment by wdconinc** [comment]
Only diff now:
image

**Review by veprbl — DISMISSED** [review]
LGTM

**Comment by veprbl** [comment]
https://github.com/eic/EICrecon/actions/runs/8855484300/job/24333764925#step:8:225
Must be due to some exception thrown.

**Comment by veprbl** [comment]
https://github.com/JeffersonLab/JANA2/issues/274

File: src/detectors/DRICH/DRICH.cc [hunk 1]
```diff
@@ -67,10 +71,30 @@ extern "C" {
...
-    track_cfg.numPlanes.insert({ "Aerogel", 5  });
-    track_cfg.numPlanes.insert({ "Gas",     10 });
+    // Track propagation
+    TrackPropagationConfig aerogel_track_cfg;
+    TrackPropagationConfig gas_track_cfg;
+
+    // get RICH geo service
+    auto richGeoSvc = app->GetService<RichGeo_service>();
+    auto dd4hepGeo = richGeoSvc->GetDD4hepGeo();
+    try {
```
Comment (wdconinc) [inline src/detectors/DRICH/DRICH.cc:81]: This needs to be an `if` with a `dd4hepGeo->detector("DRICH")` or similar call.

**Review by veprbl — APPROVED** [review]

**Pull request merged**